### PR TITLE
[BEAM-8691] bigtable-client-core dependency upgrade

### PR DESCRIPTION
**Pull request by suztomo**
https://issues.apache.org/jira/projects/BEAM/issues/BEAM-8691

- Upgrading bigtable-client-core to the latest 1.13.0
- The exposed API surface reduced. The associated test is updated.
- Because bigtable-client-core:1.13.0 introduced conflicts in Maven version range, excluding grpc-core dependencies from bigtable-client-core and google-cloud-core-grpc, so that the module uses grpc version 1.25.0 ([detail](https://issues.apache.org/jira/browse/BEAM-8691?focusedCommentId=17025456&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17025456))
- Upgrading gax to avoid linkage errors ([detail](https://issues.apache.org/jira/browse/BEAM-8691?focusedCommentId=17026017&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17026017))
- Declaring grpc-grpclb 1.25.0 to avoid linkage errors ([detail](https://issues.apache.org/jira/browse/BEAM-8691?focusedCommentId=17026077&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17026077))

# Linkage Check result

Checked beam-sdks-java-core, beam-sdks-java-io-google-cloud-platform, beam-runners-google-cloud-dataflow-java and beam-sdks-java-io-hadoop-format.

https://gist.github.com/suztomo/1bedcd7c63bc88310fff3854c19710d6

For beam-sdks-java-io-google-cloud-platform and beam-runners-google-cloud-dataflow-java
- This PR removes linkage errors on gax-httpjson-0.52.0.jar
- This PR upgrades the version of appengine-api-1.0-sdk in the errors. These are irrelevant to Beam.

Using this shell script https://gist.github.com/suztomo/ec4834151a81f8e1d26aa0afb278818b

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java [hunk 1]
```diff
@@ -61,11 +61,9 @@ public void testGcpApiSurface() throws Exception {
             classesInPackage("com.google.api.client.util"),
             classesInPackage("com.google.api.services.bigquery.model"),
             classesInPackage("com.google.auth"),
-            classesInPackage("com.google.bigtable.admin.v2"),
```
Comment (suztomo) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java]: The exposed API surface reduced.

**Comment by lukecwik** [comment]
ping for tests

**Comment by lukecwik** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by lukecwik** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
Run Spark ValidatesRunner

**Comment by lukecwik** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
@lukecwik Thank you.

**Comment by suztomo** [comment]
Java precommit failed:
```
18:48:45 > Task :sdks:java:io:rabbitmq:test FAILED
18:48:45 
18:48:45 FAILURE: Build failed with an exception.
18:48:45 
18:48:45 * What went wrong:
18:48:45 Execution failed for task ':sdks:java:io:rabbitmq:test'.
18:48:45 > Process 'Gradle Test Executor 124' finished with non-zero exit value 143
18:48:45   This problem might be caused by incorrect test process configuration.
18:48:45   Please refer to the test execution section in the User Manual at https://docs.gradle.org/5.2.1/userguide/java_testing.html#sec:test_execution
```

Java_Examples_Dataflow failed:

```
java.lang.RuntimeException: Dataflow job 2020-01-29_13_55_09-15611797686391011440 terminated in state UNRECOGNIZED but did not return a failure reason.
```

They both seem transient errors.

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
@boyuanzz Thank you!

**Comment by suztomo** [comment]
Java precommit failed:
```
WARNING: Found Netty's native epoll transport in the classpath, but epoll is not available. Using NIO instead.
java.lang.UnsatisfiedLinkError: could not load a native library: netty-transport-native-epoll

Caused by: java.lang.RuntimeException: failed to get field ID: DefaultFileRegion.transfered
```

https://builds.apache.org/job/beam_PreCommit_Java_Phrase/1686/testReport/junit/org.apache.beam.sdk.io.cassandra/CassandraIOTest/classMethod/


Java_Examples_Dataflow failed again with the same error.

```
java.lang.RuntimeException: Dataflow job 2020-01-29_20_12_07-5864892158255540617 terminated in state UNRECOGNIZED but did not return a failure reason.
```

Sent an email on this to beam dev mailing list.

**Comment by udim** [comment]
Run Java PreCommit

**Comment by udim** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
Dataflow example still fails. This time it's "No space left on device":
```
12:39:08 > Task :runners:google-cloud-dataflow-java:examples:preCommitLegacyWorker
12:39:08 java.io.IOException: No space left on device
12:39:08 com.esotericsoftware.kryo.KryoException: java.io.IOException: No space left on device
```
https://builds.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Phrase/268/consoleFull

Is this specific to this PR? I'll confirm the word-count example in my local environment with my branch.

@udim Thank you.

**Comment by udim** [comment]
> Dataflow example still fails. This time it's "No space left on device":
> 
> ```
> 12:39:08 > Task :runners:google-cloud-dataflow-java:examples:preCommitLegacyWorker
> 12:39:08 java.io.IOException: No space left on device
> 12:39:08 com.esotericsoftware.kryo.KryoException: java.io.IOException: No space left on device
> ```
> 
> https://builds.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Phrase/268/consoleFull
> 
> Is this specific to this PR? I'll confirm the word-count example in my local environment with my branch.
> 
```
jenkins@apache-beam-jenkins-7:~$ df -h .
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1       485G  484G 1001M 100% /
```
Probably not specific to your PR.

**Comment by alexvanboxel** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
R: @lukecwik @kennknowles 
21 successful checks! Would you review/merge this PR?


@alexvanboxel Thank you.

File: sdks/java/io/google-cloud-platform/build.gradle [hunk 2]
```diff
@@ -41,9 +41,14 @@ dependencies {
   compile library.java.google_auth_library_credentials
   compile library.java.google_auth_library_oauth2_http
   compile library.java.google_cloud_bigquery_storage
-  compile library.java.google_cloud_bigtable_client_core
+  compile(library.java.google_cloud_bigtable_client_core) {
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/build.gradle]: Why are these exclusions needed?

The maven pom that this module will generate has grpc-core and grpclb as top level dependencies and maven resolution rules will choose those over the ones specified by bigtable-client-core or google-cloud-core-grpc so I would have expected that we wouldn't need the exclusions.

Comment (suztomo) [reply sdks/java/io/google-cloud-platform/build.gradle]: These exclusions are for Maven's version range constraint. The dependency was transitively requesting `io.grpc:grpc-core:jar:[1.26.0,1.26.0]`, while another dependency requests `io.grpc:grpc-core:jar:[1.25.0,1.25.0]`. Detail in https://issues.apache.org/jira/browse/BEAM-8691?focusedCommentId=17025456&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17025456

Comment (lukecwik) [reply sdks/java/io/google-cloud-platform/build.gradle]: I wasn't aware that the build would fail before the dependency mediation kicked in.

**Pull request merged**